### ISSUE TITLE
Http DaprClient unable to connect to Dapr

### ISSUE THREAD

**Issue by RicardoNiepel** [dapr/java-sdk#209]
## Environment
```powershell
PS C:\Sources\dapr-java-sdk> mvn -v
  Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
  Maven home: C:\Program Files\apache-maven-3.6.3\bin\..
  Java version: 11.0.6, vendor: AdoptOpenJDK, runtime: C:\Users\riniepel\AppData\Local\Programs\AdoptOpenJDK
  Default locale: de_DE, platform encoding: Cp1252
  OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
PS C:\Sources\dapr-java-sdk>dapr --version
  CLI version: 0.3.0
  Runtime version: 0.3.0
```

**Branch:** release-0.2.0-beta

## Expected Behavior

The [Method invocation HTTP sample](https://github.com/dapr/java-sdk/tree/master/examples/src/main/java/io/dapr/examples/invoke/http) should be possible to follow and start the ```DemoService``` and ```InvokeClient``` without any errors.

## Actual Behavior

Starting the ```InvokeClient``` brings the following error

```sh
The InvokeClient was unable to connect to Dapr:
== APP == java.lang.RuntimeException: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:3006
== APP ==     at io.dapr.client.DaprHttp.lambda$invokeApi$3 (DaprHttp.java:214)
== APP ==     at reactor.core.publisher.MonoCallable.call (MonoCallable.java:91)
== APP ==     at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap (FluxFlatMap.java:126)
== APP ==     at reactor.core.publisher.MonoFlatMap.subscribeOrReturn (MonoFlatMap.java:53)
== APP ==     at reactor.core.publisher.Mono.subscribe (Mono.java:4090)
== APP ==     at reactor.core.publisher.Mono.block (Mono.java:1662)
== APP ==     at io.dapr.examples.invoke.http.InvokeClient.main (InvokeClient.java:34)
```

But I was able to send manually a POST successful to
	http://localhost:3006/v1.0/invoke/invokedemo/method/say
at the same time

## Steps to Reproduce the Problem

Start ```DemoService``` and ```InvokeClient```.

### DemoService
```
dapr run --app-id invokedemo --app-port 3000 --port 3005 -- mvn exec:java -D exec.mainClass=io.dapr.exames.invoke.http.DemoService -D exec.args="-p 3000"
Starting Dapr with id invokedemo. HTTP Port: 3005. gRPC Port: 52317
You're up and running! Both Dapr and your app logs will appear here.

== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="starting Dapr Runtime -- version 0.3.0 -- commit v0.3.0-rc.0-1-gfe6c306-dirty
"
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="log level set to: info"
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="standalone mode configured"
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="dapr id: invokedemo"
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="loaded component messagebus (pubsub.redis)"      
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="loaded component sample123 (bindings.kafka)"     
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="loaded component statestore (state.redis)"       
== DAPR == time="2020-02-05T11:42:23+01:00" level=info msg="application protocol: http. waiting on port 3000"
== APP == [INFO] Scanning for projects...
== APP == [WARNING] 
== APP == [WARNING] Some problems were encountered while building the effective model for io.dapr:dapr-sdk-examples:jar:0.2.0-beta-1 
== APP == [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-deploy-plugin is missing. @ line 130, column 15

== APP == [WARNING]
== APP == [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
== APP == [WARNING]
== APP == [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
== APP == [WARNING]
== APP == [INFO] Inspecting build with total of 1 modules...
== APP == [INFO] Installing Nexus Staging features:
== APP == [INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
== APP == Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/maven-metadata.xml
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/maven-metadata.xml (741 B at 2.0 kB/s)
== APP == Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/1.6.0/exec-maven-plugin-
1.6.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/1.6.0/exec-maven-plugin-1.6.0.pom (13 kB at 2 kB/s)
== APP == Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/1.6.0/exec-maven-plugin-
1.6.0.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/mojo/exec-maven-plugin/1.6.0/exec-maven-plugin-1.6.0.jar (58 kB at 2 kB/s)
    == APP == [INFO] 
    == APP == [INFO] ---------------------< io.dapr:dapr-sdk-examples >----------------------
== APP == [INFO] Building dapr-sdk-examples 0.2.0-beta-1
== APP == [INFO] --------------------------------[ jar ]---------------------------------
    == APP == [INFO] 
    == APP == [INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ dapr-sdk-examples ---
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-r
eporting-api-2.2.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-reporting-api-2.2.1.p (1.9 kB at 25 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting/2.2.1/maven-repor
ting-2.2.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting/2.2.1/maven-reporting-2.2.1.pom (1.4  at 22 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-sink-api/1.1/doxia-sink-api-1.1
.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-sink-api/1.1/doxia-sink-api-1.1.pom (2.0 kB at 31 ks)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia/1.1/doxia-1.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia/1.1/doxia-1.1.pom (15 kB at 211 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-a
pi-1.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-api-1.1.pom (1.6 kB a
24 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/commons-cli/commons-cli/1.2/commons-cli-1.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/commons-cli/commons-cli/1.2/commons-cli-1.2.pom (8.0 kB at 116 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.pom (11 kB at 164 kB/s)
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-r
eporting-api-2.2.1.jar
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-a
pi-1.1.jar
== APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-sink-api/1.1/doxia-sink-api-1.1
.jar
    == APP == Downloading from central: https://repo.maven.apache.org/maven2/commons-cli/commons-cli/1.2/commons-cli-1.2.jar
== APP == Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar    
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar (54 kB at 296 kB/s)
    == APP == Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-sink-api/1.1/doxia-sink-api-1.1.
jar (13 kB at 65 kB/s)
    == APP == Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-re
porting-api-2.2.1.jar (9.8 kB at 43 kB/s)
    == APP == Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-ap
i-1.1.jar (11 kB at 47 kB/s)
    == APP == Downloaded from central: https://repo.maven.apache.org/maven2/commons-cli/commons-cli/1.2/commons-cli-1.2.jar (41 kB at 128
 kB/s)
    == APP == 11:42:32.146 [io.dapr.examples.invoke.http.DemoService.main()] INFO org.springframework.core.KotlinDetector - Kotlin reflec
tion implementation not found at runtime, related features won't be available.
    == APP == 
    == APP ==   .   ____          _            __ _ _
== APP ==  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
== APP == ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
== APP ==  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
== APP ==   '  |____| .__|_| |_|_| |_\__, | / / / /
== APP ==  =========|_|==============|___/=/_/_/_/
== APP ==  :: Spring Boot ::        (v2.2.2.RELEASE)
== APP ==
    == APP == 2020-02-05 11:42:32.688  INFO 36484 --- [oService.main()] i.dapr.examples.invoke.http.DemoService  : Starting DemoService o
n MININT-DE7ORK7 with PID 36484 (C:\Sources\dapr-java-sdk\examples\target\classes started by riniepel in C:\Sources\dapr-java-sdk\examples)   
== APP == 2020-02-05 11:42:32.690  INFO 36484 --- [oService.main()] i.dapr.examples.invoke.http.DemoService  : No active profile set,set,
 falling back to default profiles: default
== APP == 2020-02-05 11:42:33.000  WARN 36484 --- [kground-preinit] o.s.h.c.j.Jackson2ObjectMapperBuilder    : For Jackson Kotlin cla cla
sses support please add "com.fasterxml.jackson.module:jackson-module-kotlin" to the classpath
== APP == 2020-02-05 11:42:34.164  INFO 36484 --- [oService.main()] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized wit wit
h port(s): 3000 (http)
== APP == 2020-02-05 11:42:34.179  INFO 36484 --- [oService.main()] o.apache.catalina.core.StandardService   : Starting service [TomcTomc
at]
== APP == 2020-02-05 11:42:34.180  INFO 36484 --- [oService.main()] org.apache.catalina.core.StandardEngine  : Starting Servlet enginngin
e: [Apache Tomcat/9.0.29]
== APP == 2020-02-05 11:42:34.307  INFO 36484 --- [oService.main()] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring emg em
bedded WebApplicationContext
== APP == 2020-02-05 11:42:34.308  INFO 36484 --- [oService.main()] o.s.web.context.ContextLoader            : Root WebApplicationConnCon
text: initialization completed in 1541 ms
orService 'applicationTaskExecutor'
== APP == 2020-02-05 11:42:35.010  INFO 36484 --- [oService.main()] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(
(s): 3000 (http) with context path ''
== APP == 2020-02-05 11:42:35.013  INFO 36484 --- [oService.main()] i.dapr.examples.invoke.http.DemoService  : Started DemoService in 
 2.8 seconds (JVM running for 11.801)
== DAPR == time="2020-02-05T11:42:35+01:00" level=info msg="application discovered on port 3000"
== APP == 2020-02-05 11:42:35.193  INFO 36484 --- [nio-3000-exec-2] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring Dis
spatcherServlet 'dispatcherServlet'
== APP == 2020-02-05 11:42:35.193  INFO 36484 --- [nio-3000-exec-2] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'd
dispatcherServlet'
== APP == 2020-02-05 11:42:35.200  INFO 36484 --- [nio-3000-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initializatio
on in 7 ms
== DAPR == time="2020-02-05T11:42:35+01:00" level=info msg="application configuration loaded"
== DAPR == 2020-02-05 11:42:35.358569 I | redis: connecting to localhost:6379
== DAPR == 2020-02-05 11:42:35.363584 I | redis: connected to localhost:6379 (localAddr: [::1]:52374, remAddr: [::1]:6379)
== DAPR == time="2020-02-05T11:42:35+01:00" level=info msg="Initialized service discovery to standalone"
== DAPR == time="2020-02-05T11:42:45+01:00" level=warning msg="failed to init output binding bindings.kafka: kafka: client has run out
t of available brokers to talk to (Is your cluster reachable?)"
== DAPR == time="2020-02-05T11:42:55+01:00" level=warning msg="failed to init input binding bindings.kafka: kafka: client has run out 
 of available brokers to talk to (Is your cluster reachable?)"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actor runtime started. actor idle timeout: 1h0m0s. actor scan interval: 30
0s"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: starting connection attempt to placement service at localhost:6050
0"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="http server is running on port 3005"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="gRPC server is running on port 52317"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="local service entry announced"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="dapr initialized. Status: Running. Init Elapsed 32085.5672ms"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: established connection to placement service at localhost:6050"    
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: placement order received: lock"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: placement order received: update"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: placement tables updated"
== DAPR == time="2020-02-05T11:42:55+01:00" level=info msg="actors: placement order received: unlock"
== APP == Server: Hi Mustermann @ 2020-02-05 11:48:15.003 and metadata: {"user-agent":"PostmanRuntime/7.19.0","host":"127.0.0.1:3000"
,"content-type":"application/json","content-length":"13","accept":"*/*","cache-control":"no-cache","postman-token":"59f2b8c8-4a88-4dff-a0d9-1f235ae94386","accept-encoding":"gzip, deflate","connection":"keep-alive"}
```

### InvokeClient

```
dapr run --port 3006 -- mvn exec:java -D exec.mainClass=io.dapr.examples.invoke.http.InvokeClient -D exeargs="'message one' 'message two'"
Starting Dapr with id Monkeypear-Oriole. HTTP Port: 3006. gRPC Port: 52400
You're up and running! Both Dapr and your app logs will appear here.

== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="starting Dapr Runtime -- version 0.3.0 -- commit v0.3.0-rc.0-1-gfe6c306-dirty
"
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="log level set to: info"
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="standalone mode configured"
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="dapr id: Monkeypear-Oriole"
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="loaded component sample123 (bindings.kafka)"
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="loaded component statestore (state.redis)"  
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="loaded component messagebus (pubsub.redis)" 
== DAPR == 2020-02-05 11:43:15.390390 I | redis: connecting to localhost:6379
== DAPR == 2020-02-05 11:43:15.399389 I | redis: connected to localhost:6379 (localAddr: [::1]:52401, remAddr: [::1]:6379)
== DAPR == time="2020-02-05T11:43:15+01:00" level=info msg="Initialized service discovery to standalone"
== APP == [INFO] Scanning for projects...
== APP == [WARNING] 
== APP == [WARNING] Some problems were encountered while building the effective model for io.dapr:dapr-sdk-examples:jar:0.2.0-beta-1 
== APP == [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-deploy-plugin is missing. @ line 130, column 15

== APP == [WARNING] 
== APP == [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
== APP == [WARNING] 
== APP == [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
== APP == [WARNING] 
== APP == [INFO] Inspecting build with total of 1 modules...
== APP == [INFO] Installing Nexus Staging features:
== APP == [INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
    == APP == [INFO] 
    == APP == [INFO] ---------------------< io.dapr:dapr-sdk-examples >----------------------
== APP == [INFO] Building dapr-sdk-examples 0.2.0-beta-1
== APP == [INFO] --------------------------------[ jar ]---------------------------------
    == APP == [INFO] 
    == APP == [INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ dapr-sdk-examples ---
    == APP == 11:43:19.675 [io.dapr.examples.invoke.http.InvokeClient.main()] DEBUG reactor.util.Loggers$LoggerFactory - Using Slf4j logg
ing framework
    == APP == [WARNING] 
    == APP == java.lang.RuntimeException: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:3006
== APP ==     at io.dapr.client.DaprHttp.lambda$invokeApi$3 (DaprHttp.java:214)
== APP ==     at reactor.core.publisher.MonoCallable.call (MonoCallable.java:91)
== APP ==     at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap (FluxFlatMap.java:126)
== APP ==     at reactor.core.publisher.MonoFlatMap.subscribeOrReturn (MonoFlatMap.java:53)
== APP ==     at reactor.core.publisher.Mono.subscribe (Mono.java:4090)
== APP ==     at reactor.core.publisher.Mono.block (Mono.java:1662)
== APP ==     at io.dapr.examples.invoke.http.InvokeClient.main (InvokeClient.java:34)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
== APP ==     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
== APP ==     at java.lang.reflect.Method.invoke (Method.java:566)
== APP ==     at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    == APP ==     at java.lang.Thread.run (Thread.java:834)
== APP == Caused by: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:3006
== APP ==     at okhttp3.internal.connection.RealConnection.connectSocket (RealConnection.kt:270)
== APP ==     at okhttp3.internal.connection.RealConnection.connect (RealConnection.kt:176)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.findConnection (ExchangeFinder.kt:236)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection (ExchangeFinder.kt:109)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.find (ExchangeFinder.kt:77)
== APP ==     at okhttp3.internal.connection.Transmitter.newExchange$okhttp (Transmitter.kt:162)
== APP ==     at okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.kt:35)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.internal.cache.CacheInterceptor.intercept (CacheInterceptor.kt:82)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.internal.http.BridgeInterceptor.intercept (BridgeInterceptor.kt:84)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept (RetryAndFollowUpInterceptor.kt:71)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
    == APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.kt:184)
== APP ==     at okhttp3.RealCall.execute (RealCall.kt:66)
== APP ==     at io.dapr.client.DaprHttp.lambda$invokeApi$3 (DaprHttp.java:196)
== APP ==     at reactor.core.publisher.MonoCallable.call (MonoCallable.java:91)
== APP ==     at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap (FluxFlatMap.java:126)
== APP ==     at reactor.core.publisher.MonoFlatMap.subscribeOrReturn (MonoFlatMap.java:53)
== APP ==     at reactor.core.publisher.Mono.subscribe (Mono.java:4090)
== APP ==     at reactor.core.publisher.Mono.block (Mono.java:1662)
== APP ==     at io.dapr.examples.invoke.http.InvokeClient.main (InvokeClient.java:34)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
== APP ==     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
== APP ==     at java.lang.reflect.Method.invoke (Method.java:566)
== APP ==     at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    == APP ==     at java.lang.Thread.run (Thread.java:834)
== APP == Caused by: java.net.ConnectException: Connection refused: connect
== APP ==     at java.net.PlainSocketImpl.waitForConnect (Native Method)
== APP ==     at java.net.PlainSocketImpl.socketConnect (PlainSocketImpl.java:107)
== APP ==     at java.net.AbstractPlainSocketImpl.doConnect (AbstractPlainSocketImpl.java:399)
== APP ==     at java.net.AbstractPlainSocketImpl.connectToAddress (AbstractPlainSocketImpl.java:242)
== APP ==     at java.net.AbstractPlainSocketImpl.connect (AbstractPlainSocketImpl.java:224)
== APP ==     at java.net.SocksSocketImpl.connect (SocksSocketImpl.java:403)
== APP ==     at java.net.Socket.connect (Socket.java:609)
== APP ==     at okhttp3.internal.platform.Platform.connectSocket (Platform.kt:127)
== APP ==     at okhttp3.internal.connection.RealConnection.connectSocket (RealConnection.kt:268)
== APP ==     at okhttp3.internal.connection.RealConnection.connect (RealConnection.kt:176)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.findConnection (ExchangeFinder.kt:236)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection (ExchangeFinder.kt:109)
== APP ==     at okhttp3.internal.connection.ExchangeFinder.find (ExchangeFinder.kt:77)
== APP ==     at okhttp3.internal.connection.Transmitter.newExchange$okhttp (Transmitter.kt:162)
== APP ==     at okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.kt:35)
    == APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
    == APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.internal.cache.CacheInterceptor.intercept (CacheInterceptor.kt:82)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.internal.http.BridgeInterceptor.intercept (BridgeInterceptor.kt:84)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept (RetryAndFollowUpInterceptor.kt:71)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:112)
== APP ==     at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.kt:87)
== APP ==     at okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.kt:184)
== APP ==     at okhttp3.RealCall.execute (RealCall.kt:66)
== APP ==     at io.dapr.client.DaprHttp.lambda$invokeApi$3 (DaprHttp.java:196)
== APP ==     at reactor.core.publisher.MonoCallable.call (MonoCallable.java:91)
== APP ==     at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap (FluxFlatMap.java:126)
== APP ==     at reactor.core.publisher.MonoFlatMap.subscribeOrReturn (MonoFlatMap.java:53)
    == APP ==     at reactor.core.publisher.Mono.subscribe (Mono.java:4090)
== APP ==     at reactor.core.publisher.Mono.block (Mono.java:1662)
== APP ==     at io.dapr.examples.invoke.http.InvokeClient.main (InvokeClient.java:34)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
== APP ==     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
== APP ==     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
== APP ==     at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
== APP ==     at java.lang.Thread.run (Thread.java:834)
== APP == [INFO] ------------------------------------------------------------------------
== APP == [INFO] BUILD FAILURE
== APP == [INFO] ------------------------------------------------------------------------
== APP == [INFO] Total time:  6.973 s
== APP == [INFO] Finished at: 2020-02-05T11:43:23+01:00
== APP == [INFO] ------------------------------------------------------------------------
== APP == [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:java (default-cli) on project dapr-sdk-examples: An
n exception occured while executing the Java class. java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:3006: Connection refused: connect -> [Help 1]
== APP == [ERROR]
== APP == [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
== APP == [ERROR] Re-run Maven using the -X switch to enable full debug logging.
== APP == [ERROR]
== APP == [ERROR] For more information about the errors and possible solutions, please read the following articles:
== APP == [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
== DAPR == time="2020-02-05T11:43:25+01:00" level=warning msg="failed to init output binding bindings.kafka: kafka: client has run out
t of available brokers to talk to (Is your cluster reachable?)"
== DAPR == time="2020-02-05T11:43:25+01:00" level=warning msg="failed to init input bindings: app channel not initialized"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actor runtime started. actor idle timeout: 1h0m0s. actor scan interval: 30
0s"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: starting connection attempt to placement service at localhost:6050
0"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="http server is running on port 3006"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="gRPC server is running on port 52400"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="local service entry announced"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="dapr initialized. Status: Running. Init Elapsed 10019.4599ms"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: established connection to placement service at localhost:6050"    
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: placement order received: lock"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: placement order received: update"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: placement tables updated"
== DAPR == time="2020-02-05T11:43:25+01:00" level=info msg="actors: placement order received: unlock"
```

**Comment by youngbupark** [comment]
I'm assuming that this is the windows specific issue. On windows, localhost resolves ipv6 address by default. Normally, it falls back to ipv4 address. based on the error, this fallback resolution did not happen.

```
== APP == java.lang.RuntimeException: java.net.ConnectException: Failed to connect to localhost/0:0:0:0:0:0:0:1:3006
```

@LMWF @artursouza once we switch from localhost to 127.0.0.1, this issue may be resolved. 

cc/ @lukekim

**Comment by yaron2** [comment]
We should make sure we switch to `127.0.0.1` everywhere, across all SDKs.
Other than avoiding issues like this, on some platforms it drives a greater performance improvement by avoiding local DNS lookups.

**Comment by artursouza** [comment]
We can make a code change for this.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by LMWF** [comment]
Should be fixed by https://github.com/dapr/java-sdk/pull/215, keeping open until validated on release branch.

**Comment by youngbupark** [comment]
@RicardoNiepel 

The fix is merged to release-branch. Please try to repro on your machine. once you confirm, we can publish this to mvn repo. 

Follow the below commands to install v0.2.0-beta-2 version java-sdk on your local machine. 
```
git checkout release-0.2.0-beta
git pull
git clean -xdf
mvn clean install
```

Then try your repro steps and check if the issue is fixed.

**Comment by RicardoNiepel** [comment]
Successfully verified the fix on **master** and **release-0.2.0-beta** branch on environment:
```ps
PS C:\Sources\dapr-java-sdk> mvn -v
  Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
  Maven home: C:\Program Files\apache-maven-3.6.3\bin\..
  Java version: 11.0.6, vendor: AdoptOpenJDK, runtime: C:\Users\riniepel\AppData\Local\Programs\AdoptOpenJDK
  Default locale: de_DE, platform encoding: Cp1252
  OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
PS C:\Sources\dapr-java-sdk>dapr --version
  CLI version: 0.3.0
  Runtime version: 0.3.0
```
